x-pack/plugins/observability_solution/slo/server/errors/handler.ts

```diff
@@ -5,20 +5,33 @@
  * 2.0.
  */
 
-import { ObservabilityError, SecurityException, SLOIdConflict, SLONotFound } from './errors';
+import { Boom, badRequest, conflict, forbidden, notFound } from '@hapi/boom';
+import { SLOError, SecurityException, SLOIdConflict, SLONotFound } from './errors';
 
-export function getHTTPResponseCode(error: ObservabilityError): number {
+function handleSLOError(error: SLOError): Boom {
   if (error instanceof SLONotFound) {
-    return 404;
+    return notFound(error.message);
   }
 
   if (error instanceof SLOIdConflict) {
-    return 409;
+    return conflict(error.message);
   }
 
   if (error instanceof SecurityException) {
-    return 403;
+    return forbidden(error.message);
   }
 
-  return 400;
+  return badRequest(error.message);
+}
+
+export async function executeWithErrorHandler(fn: () => Promise<any>): Promise<any> {
+  try {
+    return await fn();
+  } catch (error) {
+    if (error instanceof SLOError) {
+      throw handleSLOError(error);
+    }
+
+    throw error;
+  }
```

Comment:
This handler is used to wrap the various route's application services and handle any domain errors throw by them, by mapping them to their Boom equivalent. Boom error are then handled by the core router.

Ideally, the core http router would offer a way to provide a custom error handler, but changing this there would require more consideration and agreement with the core team. Which can be discussed and done outside of this PR.